libgadget/blackhole.c

```diff
@@ -371,11 +389,11 @@ blackhole_accretion_preprocess(int n, TreeWalk * tw)
 static void
 blackhole_feedback_postprocess(int n, TreeWalk * tw)
 {
-    if(BHP(n).accreted_Mass > 0)
+    int PI = P[n].PI;
+    if(BH_GET_PRIV(tw)->BH_accreted_Mass[PI] > 0)
     {
-        P[n].Mass += BHP(n).accreted_Mass;
-        BHP(n).Mass += BHP(n).accreted_BHMass;
-        BHP(n).accreted_Mass = 0;
```

Comment:
No longer need this because we reallocate accreted_Mass every step the BH is active.

However, I am a bit worried if we do accumulate this variable cross steps in corner cases.
So far it looks like that's impossible to happen to me. Did you think this through?

Reply:
I think it is impossible to accumulate it - TREEWALK_REDUCE erases the previous value, so any time the particle is active it will be accumulated anew.